File: scripts/integration/deploy.js
const Web3 = require('web3');  
const fs = require('fs');  
const solc = require('solc');  
  
// configure web3 to connect to local evmos testing environment  
const web3 = new Web3('http://127.0.0.1:8545');  
  
// read solidity contract source code  
const sourceCode = fs.readFileSync('../contract/Auction.sol', 'utf8');  
  
// compile solidity contract  
const input = {  
    language: 'Solidity',  
    sources: {  
        'Auction.sol': {  
            content: sourceCode  
        }  
    },  
    settings: {  
        outputSelection: {  
            '*': {  
                '*': ['*']  
            }  
        }  
    }  
};
  
const compiled = solc.compile(JSON.stringify(input));  
  
const contractABI = JSON.parse(compiled.contracts['Auction.sol'].Auction.abi);  
const contractByteCode = compiled.contracts['Auction.sol'].Auction.bin;  
  
// deploy contract
const Auction = new web3.eth.Contract(contractABI);  
  
// get current sender's address 
const fromAddress = web3.eth.accounts[0];  
  
// estimated gas usage  
Auction.deploy({  
    data: contractByteCode  
}).estimateGas({from: fromAddress})  
.then(gas => {  
    // send deployment transaction  
    return Auction.deploy({  
        data: contractByteCode,  
        arguments: [] // if constructor has parameters, please provide them here 
    }).send({  
        from: fromAddress,  
        gas: gas,  
        gasPrice: '20000000000' // set gas price and adjust it according to actual situation 
    });  
})  
.then(instance => {  
    console.log('Contract deployed at', instance.options.address);  
  
    // trigger event  
    const amount = web3.utils.toWei('1', 'ether'); // assuming our bid is 1 eth  
    return instance.methods.bid(amount).send({  
        from: fromAddress,  
        gas: '1000000' // set appropriate gas restrictions  
    });  
})  
.then(receipt => {  
    console.log('Bid event triggered', receipt);  
})  
.catch(error => {  
    console.error('Error deploying or bidding:', error);  
});

File: scripts/integration/deploy.py
from web3 import Web3, HTTPProvider  

from solc import compile_standard  

import json    

w3 = Web3(HTTPProvider('http://127.0.0.1:8545'))  

# compile smart contracts  

with open('../contract/Auction.sol', 'r') as file:  

    source_code = file.read()  

compiled_sol = compile_standard(  

    {  

        "language": "Solidity",  

        "sources": {"Auction.sol": {"content": source_code}},  

        "settings": {"outputSelection": {"*": {"*": ["abi", "metadata", "bin"]}}}  

    },  

    solc_version="0.8.0"  # adjust according to your solidity version 

)  

# obtain abi and binary code of contract

abi = compiled_sol['contracts']['Auction.sol']['Auction']['abi']  

bin = compiled_sol['contracts']['Auction.sol']['Auction']['bin']  

  
# deploy contract  

AuctionContract = w3.eth.contract(abi=abi, bytecode=bin)  

construct_txn = AuctionContract.constructor().buildTransaction({  

    'from': w3.eth.accounts[0],  # use first account as sender  

    'gas': 4700000,  # set gas restrictions 

    'gasPrice': w3.toWei('10', 'aunit')  # set gas price  

})  

signed_txn = w3.eth.account.signTransaction(construct_txn, private_key='YOUR_PRIVATE_KEY')  # replace with your private key  

tx_hash = w3.eth.sendRawTransaction(signed_txn.rawTransaction)  

  
# waiting for transactions to be packaged into blocks 

tx_receipt = w3.eth.waitForTransactionReceipt(tx_hash)  

contract_address = tx_receipt.contractAddress  

print(f"Contract deployed to: {contract_address}")  

  
# create contract instance
auction = w3.eth.contract(address=contract_address, abi=abi)  

# trigger bidrecord event  

amount = 1000000000000000000  # assuming we want to send 1 eth, this is 10 ^ 18 wei 

bid_txn = auction.functions.bid(amount).buildTransaction({  

    'from': w3.eth.accounts[0],  # using same account as sender  

    'gas': 4700000,  # set gas restrictions  

    'gasPrice': w3.toWei('10', 'aunit')  # set gas price  

})  

signed_bid_txn = w3.eth.account.signTransaction(bid_txn, private_key='YOUR_PRIVATE_KEY')  # replace with your private key  

bid_tx_hash = w3.eth.sendRawTransaction(signed_bid_txn.rawTransaction)  

# waiting for transactions to be packaged into blocks

w3.eth.waitForTransactionReceipt(bid_tx_hash)  

print("BidRecord event triggered successfully!")